source/extensions/common/proxy_protocol/proxy_protocol_header.cc

```diff
@@ -95,6 +101,23 @@ void generateV2Header(const std::string& src_addr, const std::string& dst_addr,
   out.add(ports, 4);
 }
 
+void generateV2Header(const Network::Address::Ip& source_address,
+                      const Network::Address::Ip& dest_address, Buffer::Instance& out) {
+  generateV2Header(source_address.addressAsString(), dest_address.addressAsString(),
+                   source_address.port(), dest_address.port(), source_address.version(), out);
+}
+
+void generateProxyProtoHeader(const envoy::config::core::v3::ProxyProtocolConfig& config,
+                              const Network::Connection& connection, Buffer::Instance& out) {
+  const Network::Address::Ip& dest_address = *connection.localAddress()->ip();
+  const Network::Address::Ip& source_address = *connection.remoteAddress()->ip();
+  if (config.version() == envoy::config::core::v3::ProxyProtocolConfig::V1) {
```

Comment:
Added the utility, which I think is 100% worthwhile especially if folks eventually make use of extensions.

one concern I noticed that I've got source/dest local/remote inverted from what Weston did in 
https://github.com/envoyproxy/envoy/pull/10682/files

I *think* this is right - we're trying to communicate about the downstream connection.  the client's destination address is the server's local address, and the client's source address is the remote address as Envoy sees it, but I'd appreciate a sanity check in case I have it backwards.

Reply:
Yeah this looks right to me. We should fix in other places if needed.

Reply:
Yup this is correct for downstream connections.

As you noticed, mine defaults to the reversed way for V1 (https://github.com/envoyproxy/envoy/pull/10682/files#diff-a25674ee929a59f2ed07b39eedfff4ccR84). This is because if there's no downstream address info (in health checks, for example), the proxy proto info is just the standard upstream connection info. If there is downstream address info available, it's sets up the src/dst the same as you have here (https://github.com/envoyproxy/envoy/pull/10682/files#diff-a25674ee929a59f2ed07b39eedfff4ccR93)